protoc-gen-hack/plugin.go

```diff
@@ -402,6 +403,37 @@ func (f field) phpType() string {
 }
 
 func (f field) defaultValue() string {
+	if f.syn == SyntaxProto2 {
+		// custom default value.
+		if dv := f.fd.GetDefaultValue(); dv != "" {
+			// proto2 custom default.
+			switch t := f.fd.GetType(); t {
+			case desc.FieldDescriptorProto_TYPE_FIXED64, desc.FieldDescriptorProto_TYPE_UINT64:
+				u64, err := strconv.ParseUint(dv, 10, 64)
+				if err != nil {
+					panic(fmt.Errorf("failed to parse custom default uint64 value: %v", err))
+				}
+				return strconv.FormatInt(int64(u64), 10)
+			case desc.FieldDescriptorProto_TYPE_STRING:
+				return "'" + strings.Replace(dv, "'", "\\'", -1) + "'"
+			case desc.FieldDescriptorProto_TYPE_BYTES:
+				// TODO, is this correct unescaping for C escaped values?
+				return "\\stripcslashes('" + dv + "')"
```

Comment:
Alternately, is it safer to write this in hex and call hex2bin?

https://stackoverflow.com/questions/49382760/php-create-a-valid-byte-string

Reply:
The contents of dv is "C escaped" bytes: https://github.com/protocolbuffers/protobuf/blob/main/src/google/protobuf/descriptor.proto#L204